### DISCUSSION TITLE
How to use with Civet?

### DISCUSSION THREAD

**Discussion by 0x241F31** [vikejs/vike#998] [Discussion]
Want to try Civet language, but seems like it doesn't work well vite-plugin-ssr:
https://github.com/lorefnon/vite-plugin-civet
```
14:38:14 web.1  | 2:38:14 PM [vite-plugin-ssr][request(1)] HTTP request: /welcome
14:38:14 web.1  | 2:38:14 PM [vite-plugin-ssr][request(1)][Wrong Usage] Vite plugins that minify the HTML are not supported by vite-plugin-ssr, see https://github.com/brillout/vite-plugin-ssr/issues/224
```
Vite plugins:
```ts
 СivetPlugin(),
 VPS({
    includeAssetsImportedByServer: true,
    prerender: {
       partial: true,
       noExtraDir: true,
    },
 }),
```

Do not have time now to investigate it myself right now.

**Comment by brillout** [comment]
It's possibly because `vite-plugin-civet` transforms HTML using `transformIndexHtml()` but I believe it should use a `transform()` hook instead. CC @lorefnon @edemaine.

**Reply by brillout** [reply]
(Btw. Civet looks really neat - [my tweet about it](https://twitter.com/brillout/status/1676531595103051778).)

**Reply by edemaine** [reply]
I've been exploring this change for Astro integration, but haven't gotten it working yet. Maybe this plugin is a simpler goal to try to integrate with. I'll give a try! (but might take a while, various other deadlines in the way)

**Reply by brillout** [reply]
👍 No rush. Keep us udpated about it. Looking forward to seeing vite-plugin-ssr users using Civet.

I hope Civet is going to become mainstream 🤞 (it's first time I say that for a JS preprocessor...).

**Comment by brillout** [comment]
> I would really like to use Civet in a project that I have just started!

I can't speak for the Civet team, but I guess PR welcome. Chances are that it's a relateively straightforward change. You can also patch Civet to disable Civet's use of `transformIndexHtml()`.

**Reply by 0x241F31** [reply]
@brillout Does VPS only looks for `.{ts,js,tsx,jsx}` pages and renderer files?

**Reply by brillout** [reply]
See `isScriptFile.ts`.
